inc/functions_user.php

```diff
@@ -173,13 +172,107 @@ function update_password($uid, $password, $salt="")
  * @param string $password The md5()'ed password.
  * @param string $salt The salt.
  * @return string The password hash.
+ * @deprecated deprecated since version 1.8.9 Please use other alternatives.
  */
 function salt_password($password, $salt)
 {
 	return md5(md5($salt).$password);
 }
 
 /**
+ * Salts a password based on a supplied salt.
+ *
+ * @param string $password The input password.
+ * @param string $salt The salt used by the MyBB algorithm.
+ * @param string $user (Optional) An array containing password-related data.
+ * @return string The password hash.
+ */
+function create_password_hash($password, $salt, $user = false)
+{
+	global $plugins;
+
+	$parameters = compact('password', 'salt', 'user');
+
+	if(!defined('IN_INSTALL') && !defined('IN_UPGRADE'))
+	{
+		$plugins->run_hooks('create_password_hash', $parameters);
+	}
+
+	if(is_string($parameters))
+	{
+		return $parameters;
```

Comment:
I still don't see why we shouldn't use the return value:

```php
$plugin_validation = false;

$plugin_validation = $plugins->run_hook(...);

if(is_string($plugin_validation))
{
    return $plugin_validation;
}
```

Even if we stay with using the parameter as return variable I'd much more prefer to have a seperate index for the return value (eg `$parameters['hash']`) as overwriting the normal parameters is something I as a plugin author would never guess without looking at the code. It simply seems wrong.